messaging/src/main/java/org/axonframework/eventhandling/tokenstore/jpa/JpaTokenStore.java

```diff
@@ -257,6 +283,59 @@ protected TokenEntry loadToken(String processorName, int segment, EntityManager
         return token;
     }
 
+    /**
+     * Tries loading an existing token owned by a processor with given {@code processorName} and {@code segment}. If such a token entry exists an attempt will
+     * be made to claim the token. If that succeeds the token will be returned. If the token is already owned by another node an {@link
+     * UnableToClaimTokenException} will be thrown.
+     * <p>
+     * If no such token exists yet, a new token entry will be inserted with {@code null} token owned by this node and return {@code null}.
+     * <p>
+     * If a token has been claimed, the {@code segment} will be validated by checking the database for the split and merge candidate segments. If a concurrent
+     * split or merge operation has been detected, the calim will be released and an {@link UnableToClaimTokenException} will be thrown.}
+     *
+     * @param processorName the name of the processor to load or insert a token entry for
+     * @param segment       the segment of the processor to load or insert a token entry for
+     * @param entityManager the entity manager instance to use for the query
+     * @return the tracking token of the fetched entry or {@code null} if a new entry was inserted
+     * @throws UnableToClaimTokenException if the token cannot be claimed because another node currently owns the token or if the segment has been split or
+     *                                     merged concurrently
+     */
+    protected TokenEntry loadToken(String processorName, Segment segment, EntityManager entityManager) {
+        TokenEntry token = loadToken(processorName, segment.getSegmentId(), entityManager);
+        try {
+            validateSegment(processorName, segment, entityManager);
+        } catch (UnableToClaimTokenException e) {
+            token.releaseClaim(nodeId);
+            throw e;
+        }
+        return token;
+    }
+
+    /**
+     * Validate a {@code segment} by checking for the existence of a split or merge candidate segment.
+     * <p>
+     * If the segment has been split concurrently, the split segment candidate will be found, indicating that we have claimed an incorrect {@code segment}. If
+     * the segment has been merged concurrently, the merge candidate segment will no longer exist, also indicating that we have claimed an incorrect {@code
+     * segment}.
+     *
+     * @param processorName the name of the processor to load or insert a token entry for
+     * @param segment       the segment of the processor to load or insert a token entry for
+     */
+    private void validateSegment(String processorName, Segment segment, EntityManager entityManager) {
+        TokenEntry mergeableSegment = entityManager //This segment should exist
+                .find(TokenEntry.class, new TokenEntry.PK(processorName, segment.mergeableSegmentId()), loadingLockMode);
+        if(mergeableSegment == null) {
```

Comment:
Nit: The indentation is slightly off here.

Reply:
Neither I, nor IntelliJ can spot the indentation error, did you mean this line, or another one?

Reply:
```suggestion
        if (mergeableSegment == null) {
```

Reply:
:point_up: extremely minor :sweat_smile: